src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -737,6 +738,52 @@ PhasePartition(ts::PhaseDry{FT}) where {FT<:Real} = PhasePartition(FT(0), FT(0),
 PhasePartition(ts::PhaseEquil) = PhasePartition_equil(air_temperature(ts), air_density(ts), ts.q_tot)
 PhasePartition(ts::PhaseNonEquil) = ts.q
 
+function roots_∂e_int_∂T(T::FT, e_int::FT, ρ::FT, q_tot::FT) where {FT<:Real}
+  cvm = cv_m(PhasePartition(q_tot))
```

Comment:
This seems odd. `cv_m` needs to depend on the equilibrium partitioning of `q_tot` into `q_l` and `q_i`, which depends on the input temperature `T`, but `T` is not an input here. It seems to me you need
```suggestion
  cvm = cv_m(PhasePartition_equil(T, ρ, q_tot))
```
here.